homeassistant/components/mqtt/config_flow.py

```diff
@@ -564,58 +568,24 @@ async def async_step_hassio_confirm(
 class MQTTOptionsFlowHandler(OptionsFlow):
     """Handle MQTT options."""
 
-    def __init__(self) -> None:
-        """Initialize MQTT options flow."""
-        self.broker_config: dict[str, Any] = {}
-
     async def async_step_init(self, user_input: None = None) -> ConfigFlowResult:
         """Manage the MQTT options."""
-        return await self.async_step_broker()
-
-    async def async_step_broker(
-        self, user_input: dict[str, Any] | None = None
-    ) -> ConfigFlowResult:
-        """Manage the MQTT broker configuration."""
-        errors: dict[str, str] = {}
-        fields: OrderedDict[Any, Any] = OrderedDict()
-        validated_user_input: dict[str, Any] = {}
-        if await async_get_broker_settings(
-            self,
-            fields,
-            self.config_entry.data,
-            user_input,
-            validated_user_input,
-            errors,
-        ):
-            self.broker_config.update(
-                update_password_from_user_input(
-                    self.config_entry.data.get(CONF_PASSWORD), validated_user_input
-                ),
-            )
-            can_connect = await self.hass.async_add_executor_job(
-                try_connection,
-                self.broker_config,
-            )
-
-            if can_connect:
-                return await self.async_step_options()
-
-            errors["base"] = "cannot_connect"
-
-        return self.async_show_form(
-            step_id="broker",
-            data_schema=vol.Schema(fields),
-            errors=errors,
-            last_step=False,
-        )
+        return await self.async_step_options()
 
     async def async_step_options(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Manage the MQTT options."""
         errors = {}
-        current_config = self.config_entry.data
-        options_config: dict[str, Any] = {}
+
+        # Can be removed when config entry is bumped to version 2.1
+        # with HA Core 2026.1.0
+        legacy_data_options = {
+            key: self.config_entry.data[key]
+            for key in ENTRY_OPTION_FIELDS
+            if key in self.config_entry.data
+        }
+        options_config: dict[str, Any] = self.config_entry.options | legacy_data_options
```

Comment:
Why do we need this, we move the data in the migrator function?

Reply:
removed